neuromation/api/jobs.py

```diff
@@ -162,14 +171,43 @@ def __init__(self, core: _Core, config: _Config) -> None:
             # an error is raised for status >= 400
             return None  # 201 status code
 
-    async def save(self, id: str, image: RemoteImage) -> None:
+    async def save(
+        self,
+        id: str,
+        image: RemoteImage,
+        *,
+        progress: Optional[AbstractDockerImageProgress] = None,
+    ) -> RemoteImage:
         if not _is_in_neuro_registry(image):
             raise ValueError(f"Image `{image}` must be in the neuromation registry")
+
+        if progress is None:
+            progress = _DummyProgress()
+        progress.commit(JobProgressCommit(id, image))
+        is_committing = True
+        is_pushing = False
+
         payload = {"container": {"image": _as_repo_str(image)}}
         url = self._config.cluster_config.monitoring_url / f"{id}/save"
-        async with self._core.request("POST", url, json=payload):
-            # an error is raised for status >= 400
-            return None  # 201 status code
+        async with self._core.request("POST", url, json=payload) as resp:
+            async for chunk in resp.content:
```

Comment:
Please replace this loop with two sequential ones: first for committing, next for pushing.

Reply:
OK thanks

Reply:
How should it look like? if `self._core.request("POST", url, json=payload)` returns `asynccontextmanager`, which streams the following messages:
```
{"status": "Committing container DIGEST as image alpine:latest"},
        {"status": "Committed"},
        {"status": "The push refers to repository [localhost:5000/alpine]"},
        {"status": "Preparing", "progressDetail": {}, "id": "a31dbd3063d7"},
        {
            "status": "Pushing",
            "progressDetail": {"current": 3584},
            "progress": " 3.584kB",
            "id": "0acd017a4b67",
        },
        {"status": "Pushed", "progressDetail": {}, "id": "0acd017a4b67"},
        {"status": "job-id: digest: sha256:DIGEST size: 1359"},
        {
            "progressDetail": {},
            "aux": {"Tag": "job-id", "Digest": "sha256:DIGEST", "Size": 1359},
        },
```

Reply:
So you effectively have something like:
```
line1 = await resp.content.readline()  # Committing container
line2 = await resp.content.readline()  # Committed
line3 = await resp.content.readline()  # start pushing
async for line in resp.content:
    # preparing, pushing, pushed etc
```

Reply:
achso, okay thanks!